docs/proposals/mcs/remove-derived-mcs.md

```diff
@@ -0,0 +1,352 @@
+
+### 2) Karmada adaptation
+
+What needs to be done in the `karmada` part is to fill in the `ips` field of the `ServiceImport` which imported to the member cluster on the existing basis and correspond to the `clusterIP` of `derived-foo`. Its flow diagram is as follows:
+
+![remove-derived-in-controller](./remove-derived-in-controller.jpg)
+
+As shown in the figure above, we still take the foo service as an example. After configuring coredns, we add the logic to obtain the derived-foo service in the cluster members in the ServiceImport-controller of the control plane.
+
+1. Wait for cluster member config the `derived-foo` service's `clusterIP` .
+
+2. The controller of the control plane obtains the value, and fills the value into the work corresponding to the `ServiceImport` of the control plane.
+
+3. According to the synchronization principle of `Karmada`, the `ServiceImport` object in the member cluster will be filled with the `clusterIP` value of `derived-foo`.
+
+That is, the effect of `ServiceImport` after completion needs to be:
+
+```yaml
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ServiceImport
+metadata:
+   name: foo
+   namespace: default
+spec:
+   type: ClusterSetIP
+   ports:
+   - name: "http"
+     port: 80
+     protocol: TCP
+   ips:
+   - "10.10.0.5" # clusterIP for cross-cluster access
+```
+
+### 3) Suggested operation
+
+The recommended configuration instructions for the `coreDNS`+`multicluster` plug-in solution are given below.
+
+#### Environment preparation:
+
+A normal Karmada cluster, and can perform existing mcs operations.
+
+The following uses the cluster member2 where the client resides as an example to access the `foo`service using the suffix `clusterset.local`. This solution needs to be configured as follows:
+
+#### coredns compile
+
+Refer to [Compile Instructions](#Compilation Instructions).
+
+#### coredns configuration
+
+Refer to [Configuration Description](#Configuration Description).
+
+#### Client pod configuration
+
+Taking yaml as an example, you need to add the configuration item searches of dnsConfig in yaml, and set the value to `clusterset.local`. A simple and complete example is as follows:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+   name: dns-test
+spec:
+   replicas: 2
+   selector:
+     matchLabels:
+       app: dns-test
+   template:
+     metadata:
+       labels:
+         app: dns-test
+     spec:
+       containers:
+       - name: dns-test
+         image: ubuntu:18.04
+         command:
+         - "/bin/bash"
+         - "-c"
+         args: [ "while true; do sleep 3; cat /etc/resolv.conf; done" ]
+       dnsPolicy: ClusterFirst
+       dnsConfig:
+         searches:
+           -clusterset.local
+         options:
+           - name: timeout
+             value: "2
+```
+
+### Test Plan
+
+- All current testing should be passed, no break change would be involved by this feature.
+- Add new E2E tests to cover the feature, the scope should include:
```

Comment:
E2E testing is not easy to add, right?

Reply:
Yes, it's difficult. I haven't figured it out yet.